packages/internals/src/engine-commands/formatSchema.ts

```diff
@@ -1,52 +1,126 @@
 import Debug from '@prisma/debug'
-import { BinaryType } from '@prisma/fetch-engine'
-import execa from 'execa'
 import fs from 'fs'
+import { match } from 'ts-pattern'
 
-import { resolveBinary } from '../resolveBinary'
+import { ErrorArea, RustPanic } from '../panic'
+import { prismaFmt } from '../wasm'
 
 const debug = Debug('prisma:formatSchema')
 
-const MAX_BUFFER = 1_000_000_000
+type FormatSchemaParams = { schema: string; schemaPath?: never } | { schema?: never; schemaPath: string }
 
-// can be used by passing either
-// the schema as a string
-// or a path to the schema file
-export async function formatSchema({ schema }: { schema: string })
-export async function formatSchema({ schemaPath }: { schemaPath: string })
-export async function formatSchema({ schemaPath, schema }: { schemaPath?: string; schema?: string }): Promise<string> {
+function isSchemaOnly(schemaParams: FormatSchemaParams): schemaParams is { schema: string; schemaPath?: never } {
+  return !!schemaParams.schema
+}
+
+function isSchemaPathOnly(schemaParams: FormatSchemaParams): schemaParams is { schema?: never; schemaPath: string } {
+  return !!schemaParams.schemaPath
+}
+
+/**
+ * Can be used by passing either the `schema` as a string, or a path `schemaPath` to the schema file.
+ * Currently, we only use `schemaPath` in the cli. Do we need to keep supporting `schema` as well?
+ */
+export async function formatSchema(
+  { schemaPath, schema }: FormatSchemaParams,
+  inputFormattingOptions?: Partial<DocumentFormattingParams['options']>,
+): Promise<string> {
   if (!schema && !schemaPath) {
     throw new Error(`Parameter schema or schemaPath must be passed.`)
   }
 
-  const prismaFmtPath = await resolveBinary(BinaryType.prismaFmt)
-  const showColors = !process.env.NO_COLOR && process.stdout.isTTY
+  if (process.env.FORCE_PANIC_PRISMA_FMT) {
+    handleFormatPanic(
+      () => {
+        prismaFmt.debug_panic()
+      },
+      { schemaPath, schema } as FormatSchemaParams,
+    )
+  }
+
+  const schemaContent = match({ schema, schemaPath } as FormatSchemaParams)
+    .when(isSchemaOnly, ({ schema: _schema }) => {
+      debug('using a schema string directly')
+      return _schema
+    })
+    .when(isSchemaPathOnly, ({ schemaPath: _schemaPath }) => {
+      debug('reading a schema from a file')
```

Comment:
Do we really need these 2 debug? they seem not so useful

Reply:
It seemed relevant to track as one branch may trigger and error and the error doesn't, considering we also use `debug` in smaller-scope utiities like `plusX` around the codebase :)

But if it just adds noise in your opinion, we can get rid of it

Reply:
It only changes the input here, and it will be obvious what's happening when something breaks I think, the only difference is a `fs.readFileSync` when using a `schemaPath`.

Reply:
Ok, applied suggestion in latest commit